README.md

```diff
@@ -64,3 +64,7 @@ npm run build
 - Checkout Blocks + Subscriptions support
 - Support for the WooCommerce Payments gateway
 - Privacy/GDPR exporters for Subscriptions
+
+
+## Docs
+- [Deprecation guide: When and how to deprecate hooks and non-private functions in subscriptions-core](docs/deprecation-guide.md)
```

Comment:
Can we generalise this? E.g. deprecation principles or guidelines in general. 

Ideally we'd have one set of guidelines / process against all subs-family code bases, or even WCPay too.